src/openrct2/management/Research.cpp

```diff
@@ -914,50 +844,21 @@ void research_fix()
 
 void research_items_make_all_unresearched()
 {
-    rct_research_item *researchItem, *nextResearchItem, researchItemTemp;
-
-    int32_t sorted;
-    do
+    for (auto it = gResearchItemsInvented.begin(); it != gResearchItemsInvented.end();)
     {
-        sorted = 1;
-        for (researchItem = gResearchItems; researchItem->rawValue != RESEARCHED_ITEMS_SEPARATOR; researchItem++)
-        {
-            if (research_item_is_always_researched(researchItem))
-                continue;
-
-            nextResearchItem = researchItem + 1;
-            if (nextResearchItem->rawValue == RESEARCHED_ITEMS_SEPARATOR
-                || research_item_is_always_researched(nextResearchItem))
-            {
-                // Bubble up always researched item or separator
-                researchItemTemp = *researchItem;
-                *researchItem = *nextResearchItem;
-                *nextResearchItem = researchItemTemp;
-                sorted = 0;
-
-                if (researchItem->rawValue == RESEARCHED_ITEMS_SEPARATOR)
-                    break;
-            }
-        }
-    } while (!sorted);
+        auto& researchItem = *it;
+        gResearchItemsUninvented.push_back(researchItem);
+        gResearchItemsInvented.erase(it);
+    }
 }
 
 void research_items_make_all_researched()
 {
-    rct_research_item *researchItem, researchItemTemp;
-
-    // Find separator
-    for (researchItem = gResearchItems; researchItem->rawValue != RESEARCHED_ITEMS_SEPARATOR; researchItem++)
-    {
-    }
-
-    // Move separator below all items
-    for (; (researchItem + 1)->rawValue != RESEARCHED_ITEMS_END; researchItem++)
+    for (auto it = gResearchItemsUninvented.begin(); it != gResearchItemsUninvented.end();)
```

Comment:
```suggestion
    gResearchItemsInvented = std::move(gResearchItemsUninvented);
```

Reply:
Won't that overwrite the existing contents of gResearchItemsInvented?

Reply:
Yeah I mixed this up slightly